src/renderer/utils/__tests__/jsonPath.test.tsx

```diff
@@ -0,0 +1,35 @@
+import { parseJsonPath } from "../jsonPath";
+
+describe("parseJsonPath", () => {
```

Comment:
some more test cases that I have thought of:

1. Include one with `\\\` signifying an escaped '\' or something similar.
2. perhaps escaping other characters too, like `\'` (which might break your index conversion)